src/sagemaker/jumpstart/hub/parser_utils.py

```diff
@@ -29,20 +33,29 @@ def snake_to_upper_camel(snake_case_string: str) -> str:
     return upper_camel_case_string
 
 
-def walk_and_apply_json(json_obj: Dict[Any, Any], apply) -> Dict[Any, Any]:
-    """Recursively walks a json object and applies a given function to the keys."""
+def walk_and_apply_json(
+    json_obj: Dict[Any, Any], apply, stop_keys: Optional[List[str]] = ["metrics"]
+) -> Dict[Any, Any]:
+    """Recursively walks a json object and applies a given function to the keys.
+
+    stop_keys (Optional[list[str]]): List of field keys that should stop the application function.
+        Any children of these keys will not have the application function applied to them.
+    """
 
     def _walk_and_apply_json(json_obj, new):
         if isinstance(json_obj, dict) and isinstance(new, dict):
             for key, value in json_obj.items():
                 new_key = apply(key)
-                if isinstance(value, dict):
-                    new[new_key] = {}
-                    _walk_and_apply_json(value, new=new[new_key])
-                elif isinstance(value, list):
-                    new[new_key] = []
-                    for item in value:
-                        _walk_and_apply_json(item, new=new[new_key])
+                if (stop_keys and new_key not in stop_keys) or stop_keys is None:
```

Comment:
nit: 
```
if stop_keys and new_key in stop_keys:
    new[new_key] = value
else:
    ...
```

Reply:
+1, conditional is too complex. If you want to keep this condition, create a utils and unit-test separately please.